sdk/ai/ai-projects/src/agents/assistants.ts

```diff
@@ -20,80 +21,85 @@ enum Tools {
 
 /** Creates a new agent. */
 export async function createAgent(
-    context: Client,
-    options: CreateAgentParameters,
-  ): Promise<AgentOutput> {
-    validateCreateAgentParameters(options);
-    const result = await  context.path("/assistants").post(options);
-    if (!expectedStatuses.includes(result.status)) {
+  context: Client,
+  options: CreateAgentParameters,
+): Promise<AgentOutput> {
+  validateCreateAgentParameters(options);
+  return TracingUtility.withSpan("CreateAgent", options,
+    async (updatedOptions) => {
+      const result = await context.path("/assistants").post(updatedOptions);
+      if (!expectedStatuses.includes(result.status)) {
         throw createRestError(result);
-    }
-    return result.body; 
-  }
+      }
+      return result.body as AgentOutput;
```

Comment:
Hopefully the `as AgentOutput` is still not needed. At least with `tracingClient` the typing should work but I have not looked at your wrapper yet

Reply:
Yes its not needed, will remove it.